src/main/java/edu/wpi/grip/core/events/ErrorWitness.java

```diff
@@ -0,0 +1,62 @@
+package edu.wpi.grip.core.events;
+
+import com.google.common.eventbus.EventBus;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Witness and reports errors.
+ * {@link #flagError} should be used to flags the witness that an flagError has has occurred.
+ * This will post an {@link ErrorEvent} to the {@link EventBus}
+ */
+public final class ErrorWitness {
+    private final EventBus eventBus;
+    private final Object origin;
+    private final AtomicBoolean isErrorState = new AtomicBoolean(false);
+
+    public ErrorWitness(final EventBus eventBus, final Object origin) {
+        this.eventBus = checkNotNull(eventBus, "The event bus can not be null");
+        this.origin = checkNotNull(origin, "The origin can not be null");
+    }
+
+    /**
+     * Indicates to the witness that an flagError has occurred. This will also post an {@link ErrorEvent} to the {@link EventBus}
+     *
+     * @param throwable The flagError that this is reporting
+     * @param message   Any additional details that should be associated with this message.
+     */
+    public final void flagError(final Throwable throwable, final String message) {
+        isErrorState.set(true);
+        this.eventBus.post(new ErrorEvent(origin, throwable, message));
```

Comment:
Should this only be posted if the throwable's stack trace or message changes?
That way the `ErrorEvent` will only be posted once per error instead of being posted at 30fps for steps/sources.